### DISCUSSION TITLE
Need a bit of help with rules and multiple triggers (or some other wat to make things work)

### DISCUSSION THREAD

**Discussion by noromamai** [arendst/Tasmota#22595] [Q&A] [answered]
Ok. So I need a power switch to turn itself off and back on every 15 minutes between certain times on certain days (2 options)

So far I have the below rule which works. It will turn off and back on between certain times on a day:

```
#run on tuesday, wednesday and thursday
Rule1
  on time#minute<840 do break 
  on time#minute>=1435 do break 
  on time#minute|15 do backlog power1 0; delay 5; power1 1 endon
  
#only run on friday and saturday
Rule2
  on time#minute<600 do break
  on time#minute>=1435 do break
  on time#minute|15 do backlog power1 0; delay 5; power1 1 endon
  
#run on all days
Rule3
  on time#minutes>=1438 do power1 0 endon
```

The problem now is that I want to use several timers to trigger rule1 and rule2 on the days I need them. But if I use the timer trigger Clock#Timer=<x>, I can no longer use the time#minute trigger, or do I just simply add the timer != trigger at the top of the rule? does it keep checking if the timer setting is valid or does it only trigger once when the timer time has been reached?

```
Rule1
  on clock#timer!=1 do break
  on time#minute<840 do break 
  on time#minute>=1435 do break 
  on time#minute|15 do backlog power1 0; delay 5; power1 1 endon
```

First time tasmota rules user here, any help or advice will be appreciated.

**Comment by sfromis** [comment]
Using `break` can only work when the triggers are from the **same** event (like a sensor payload). As `Clock#Timer` and `Time#Minute` fire independently of each other, with no concept like checking unchanged states, you cannot combine logic that way.

Instead of this:
```
  on time#minute<840 do break 
  on time#minute>=1435 do break 
  on time#minute|15 do backlog power1 0; delay 5; power1 1 endon
```
It could work with:
```
on Time#Minute|15 do if ((time>=840) and (time<1435)) TimedPower1 500,off endif endon
```
Then you could expand that to have the `Clock#Timer` triggers set a variable for the desired state, and expand the `if` condition to test this variable, to tickle power or not.

**Comment by noromamai** [comment, marked as answer]
What you proposed needs the Scripting Language enabled in the firmware, right? I want to keep it simple. I have implemented the following but still need to see if it actually works.

```
Rule1
  on clock#timer=1 do Rule2 1 endon
  on clock#timer=2 do Rule2 1 endon
  on clock#timer=3 do Rule2 0 endon

Rule2
  on time#minute|15 do backlog power1 0; delay 5; power1 1 endon
```
Then I have set the following timers:

Timer 1
triggers on Tue, Wed, Thu at 14:00

Timer 2
triggers on Fri, Sat at 09:30

Timer 3
triggers every day at 00:30

This works. I forgot to enable the timer to repeat which left me scratching my head the first two days. But now everything is working as intended. I have an additional timer 4 that will turn the switch off at 01:00 that does not use any rules, just a simple off.

**Reply by sfromis** [reply]
No, there was nothing using the scripter or even any custom build. Just standard rules, with a more compact way of doing the same.
